sw/device/lib/dif/dif_uart.c

```diff
@@ -110,7 +110,7 @@ static bool uart_configure(const dif_uart_t *uart,
 
   // Requested baudrate is too high for the given clock frequency.
   if (nco != nco_masked) {
-    return false;
+    return kDifUartConfigBadConfig;
```

Comment:
I think it might be nicer if this was split out to not just be "this config was bad" but maybe something more friendly like "that NCO is not supported". What do you think?

Reply:
Sure, I am all for it. I had similar suggestion on DIF SPI review, but I think guys decided that for now it's okay to have more generic error codes.

I agree and will add additional error code.